### ISSUE TITLE
PyramidScheme for DGGS level computations

### ISSUE THREAD

**Issue by felixcremer** [JuliaDataCubes/PyramidScheme.jl#28]
We can reuse the PyramidScheme code to also compute the pyramids of a DGGS dataset for that we would need a way to specify the used internal kernel for the computation either as a user input or via a keyword. 

@danlooo

**Comment by danlooo** [comment]
Yes, I think a generalization of pyramid building would allow to implement any convolution, including those involved in aggragating DGGS cells.
In the latter case, one can define this by params Kernel $K$ and Stride $S$:

### Kernel

![image](https://github.com/JuliaDataCubes/PyramidScheme.jl/assets/5780565/5eecc97e-a31d-420f-ae87-ba4fd08ee35b)

$$
K=\frac{1}{8}*
\begin{matrix}
0 & 1 & 1\\
1 & 2 & 1\\
1 & 1 & 0
\end{matrix}
$$

in which $K$ weights the individual child cell values by area. However, one need to handle edge cases e.g. 12 pentagons.

### Stride

![image](https://github.com/JuliaDataCubes/PyramidScheme.jl/assets/5780565/62f36538-db61-46a7-ba10-9e1bbe92c809)

Stride $S$ is $2*cell width$ in both $j$ and $i$ direction. Overlap to other faces of the icosahedron might be resolved by adding a paddings to the data.
